docs/adrs/minimal-lifecycle.md

```diff
@@ -0,0 +1,137 @@
+---
+title: "minimal channel life-cycle"
+status: proposed
+authors: "@waalge"
+date: 2024-10-06
+tags:
+  - lifecycle
+---
+
+## Context
+
+CL needs a starting point. We propose the minimal life-cycle, establishing
+fundamental components and introducing the vocab. This is predominantly focused
+on the essential parts of the L1 component - the MVP.
+
+## Decision
+
+### Overview
+
+CL is a network.
+
+Users of the network are **participants** and form the nodes of the graph of the
+network.
+
+A **channel** is an edge to the network connecting two participants. A channel
+compromises of state and actions that happen both on-chain and off-chain.
+
+On-chain, A channel is represented by utxos, at most one at tip at any one time.
+While such a utxo exists at tip, the channel is **staged** (see 'stages' below).
+After the utxo is spent and not replaced, the channel is **unstaged**. A channel
+that is unstaged will never again be staged. Unless otherwise specified,
+"channel" generally means "channel that is staged".
+
+A channel has state on the L1. Note that we avoid using the term "state" to have
+a specified meaning. This is so it can be employed and specified in a given
+context, without being confusing. The relevant state may be the datum, or the
+datum and value, or the whole utxo, or a set of utxos, or include additional
+off-chain data _etc_.
+
+The Cardano unit of change is the transaction. A transaction involving a channel
+is said to **step** the channel. A single transaction may involve many channels.
+From the perspective of a single channel, we can abstract transactions and talk
+in terms of steps.
+
+The term step is used both to refer to a specific event, or a general family of
+events. For example: "this transaction steps that channel", and "an 'open' is
+the initial step of a channel".
+
+A staged channel will be at a **stage** in its life cycle.
+
+The **opened** stage is the main stage of the channel. During this stage
+participants are actively transacting with each other off-chain.
+
+The **closed** stage is the stage following opened. The stage occurs once one of
+the participants no longer wants to continue the channel.
+
+The **resolved** stage follows the closed stage. The stage occurs once both
+participants have communicated to the L1 the conclusion of their off-chain
+transacting. This is the final stage, and afterwards the channel is unstaged.
+
+A step may:
+
+- stage or unstage a channel,
+- take a channel from a stage to the same stage, or
+- take a channel from one stage to the next stage
+
+```mermaid
+---
+title: Minimal channel life-cycle diagram
+---
+stateDiagram-v2
+    [*] --> Opened : open
+    Opened --> Closed : close
+    Closed --> Resolved : resolve
+    Resolved --> [*] : end
+    Closed --> [*] : drain
+```
+
+The **open** stages the channel to its opened stage. The participant locks funds
+that they wish to back their off-chain transacting. The two participants can now
+safely transact off-chain.
+
```

Comment:
What confuses me about this ADR is that it does not include `mutual-close`. Am I right? What is the rationale for this omission?

Reply:
It is omitted. 
mutualClose may be a convenience but is not necessary. 
Its a collaborative tx (see note). 

Its not necessary for MVP - as in there is a 'complete' (see caveat) set of functionality allowing full channel life-cycle that does not include mutualClose. Using mutualClose cannot replace close+resolve+end/ close+drain since it is not safe